### PR TITLE
update buildkit to merge sha b4ccdaf100c3093bf0d4b430a70ec7f541d25a4a

### PR DESCRIPTION
**Pull request by r-LaForge**
Linked to buildkit PR: https://github.com/earthly/buildkit/pull/99
required fsutil update: https://github.com/earthly/fsutil/pull/9

### CODE REVIEW

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -39,11 +42,8 @@ import (
 	"github.com/earthly/earthly/util/syncutil/semutil"
 	"github.com/earthly/earthly/variables"
 	"github.com/earthly/earthly/variables/reserved"
-
```
Comment (alexcb) [inline earthfile2llb/converter.go:42]: we try to keep all earthly imports separate from third party libraries -- can you restore this newline.

File: earthfile2llb/converter.go [hunk 2]
```diff
@@ -1356,7 +1356,7 @@ func (c *Converter) Healthcheck(ctx context.Context, isNone bool, cmdArgs []stri
 		return err
 	}
 	c.nonSaveCommand()
-	hc := &dockerfile2llb.HealthConfig{}
+	hc := &dockerimage.HealthConfig{}
```
Comment (alexcb) [inline earthfile2llb/converter.go:1359]: is this change required?

Comment (alexcb) [reply earthfile2llb/converter.go:1359]: nevermind.... somehow I misread this as `earthfile2llb` :)

**Review by alexcb — CHANGES_REQUESTED** [review]
it doesn't look like `ARG BUILDKIT_BASE_IMAGE=github.com/earthly/buildkit:9c256c19c5397aee1ba0379590d685c2e87ff17c+build` was updated to reflect the new buildkit git sha; did you run `earthly +update-buildkit --BUILDKIT_GIT_ORG=<git-user-or-org> --BUILDKIT_GIT_SHA=<40-char-git-reference-here>` ?

**Comment by r-LaForge** [comment]
@alexcb That's weird... 
Yep, I ran `earthly +update-buildkit --BUILDKIT_GIT_SHA=13e1ae7c49327f4c80c13d31a550de1271fa8667`. 
I updated it manually now.

**Comment by r-LaForge** [comment]
With the help of @alexcb  I've updated the script to run sed differently depending on if it is Darwin or not.

File: go.mod [hunk 3]
```diff
@@ -96,6 +97,6 @@ replace (
 	github.com/earthly/earthly/ast => ./ast
 	github.com/jessevdk/go-flags => github.com/alexcb/go-flags v0.0.0-20210722203016-f11d7ecb5ee5
 
-	github.com/moby/buildkit => github.com/earthly/buildkit v0.0.1-0.20220913002628-9c256c19c539
-	github.com/tonistiigi/fsutil => github.com/earthly/fsutil v0.0.0-20220719234708-392da7eebeb5
+	github.com/moby/buildkit => github.com/earthly/buildkit v0.0.1-0.20221026205052-75bffa4fcb41
+	github.com/tonistiigi/fsutil => github.com/earthly/fsutil v0.0.0-20221019181234-420dfa996232
```
Comment (alexcb) [inline go.mod:101]: this seems off; fsutils is at `b994beadd44391cd559120cd153f2b7de68d14c6`

File: buildkitd/Earthfile [hunk 4]
```diff
@@ -91,4 +91,12 @@ update-buildkit-earthfile:
     LOCALLY
     ARG --required BUILDKIT_GIT_SHA
     ARG --required BUILDKIT_GIT_ORG
-    RUN sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit\).*/\1:'$BUILDKIT_GIT_SHA'+build/g' Earthfile
+    # We run the Mac version on Darwin as it is not compatible with Linux sed
+    RUN if [ "$(uname)" = "Darwin" ]; then \
+            sed -E -i '' -e 's/(^[ \t]+ARG BUILDKIT_BASE_IMAGE).*\/buildkit.*/\1=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit:'$BUILDKIT_GIT_SHA'+build/g' Earthfile; \
+        else \
+            "sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE\)=github.com\/.*\/buildkit.*/\1=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit:'$BUILDKIT_GIT_SHA'+build/g' Earthfile"; \
```
Comment (alexcb) [inline buildkitd/Earthfile:98]: why is there a `"` before `sed`?

furthermore, this regex seems to be different compared with the old one -- is this intentional?

File: buildkitd/Earthfile [hunk 5]
```diff
@@ -91,4 +91,12 @@ update-buildkit-earthfile:
...
+        fi
+false:
```
Comment (alexcb) [inline buildkitd/Earthfile:100]: why is this here?

File: earthfile2llb/converter.go (hunk 1 above)
Comment (alexcb) [reply earthfile2llb/converter.go:42]: @r-LaForge did you back this change out? I'm still seeing it removed in the latest PR.

File: buildkitd/Earthfile (hunk 5 above)
Comment (r-LaForge) [reply buildkitd/Earthfile:100]: 🤦 oops

File: go.mod (hunk 3 above)
Comment (r-LaForge) [reply go.mod]: Good catch!
I must have forgotten to update the SHA to the `earthly-main` merge commit.
This has been fixed now.

File: buildkitd/Earthfile (hunk 4 above)
Comment (r-LaForge) [reply buildkitd/Earthfile]: Removed quotes.

The regex is intentionally different. Unfortunately, I was unable to get a regex that worked on both systems. They should be functionally equivalent though.

File: earthfile2llb/converter.go [hunk 6]
```diff
@@ -18,6 +18,9 @@ import (
 	"strings"
 	"time"
 
+	"github.com/alessio/shellescape"
```
Comment (alexcb) [inline earthfile2llb/converter.go:21]: can you move these to the next section down.

File: buildkitd/Earthfile (hunk 4 above)
Comment (alexcb) [reply buildkitd/Earthfile]: I understand that the MacOS regex is different; however don't understand why the sed command for linux needs to change.

File: earthfile2llb/converter.go (hunk 1 above)
Comment (r-LaForge) [reply earthfile2llb/converter.go]: I think I have the ordering we want now.

File: buildkitd/Earthfile (hunk 4 above)
Comment (r-LaForge) [reply buildkitd/Earthfile]: Oh, it changed because we couldn't change the ORG using that version.
the old one was:
```
sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit\).*/\1:'$BUILDKIT_GIT_SHA'+build/g' Earthfile
```
By including `$BUILDKIT_GIT_ORG` it only matched if the organization was the same.

File: earthfile2llb/converter.go (hunk 6 above)
Comment (r-LaForge) [reply earthfile2llb/converter.go]: done

File: buildkitd/Earthfile (hunk 4 above)
Comment (alexcb) [reply buildkitd/Earthfile]: oh sweet! that makes sense. nice change.

**Review by alexcb — APPROVED** [review]
:shipit: :tada: :taco:
